lib/virt_autotest/utils.pm

```diff
@@ -105,23 +105,49 @@ sub is_xen_host {
     return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen") || check_var("REGRESSION", "xen-hypervisor");
 }
 
-#check host to make sure it works well
+#check kernels by grep keywords from journals
 #welcome everybody to extend this function
-sub check_host {
+sub check_failures_in_journal {
+    my $machine //= 'localhost';
+
+    my $all_failures = "";
+    my @warnings = ('Started Process Core Dump', 'Call Trace');
+    foreach my $warn (@warnings) {
+        my $cmd = "journalctl | grep '$warn'";
+        $cmd = "ssh root\@$machine " . "\"$cmd\"" if $machine ne 'localhost';
+        if (script_run($cmd) == 0) {
+            record_soft_failure("$warn on $machine: bsc#5555 " . get_var('KNOWN_KERNEL_BUG'));
+            $all_failures .= "$warn on $machine: " . get_var('KNOWN_KERNEL_BUG');
```

Comment:
Hi Julie, there seems no comparison between the found warning info and the KNOWN_KERNEL_BUG in code, or  indication info to reviewers that they should compare by themselves to see if they are KNOWN bugs or they should open NEW bugs. Shall we pick either way? Otherwise, I am afraid new joiner would not know what to do with such soft_fail.

Secondly, bsc#5555 should be a fake bug id, right? Would you please adding `(fake)` after 5555?

Last, I remember we talked about how to make such record_soft_fail different from REAL soft_fail? By adding record_info (red borders) around such soft_fail?

Reply:
> Hi Julie, there seems no comparison between the found warning info and the KNOWN_KERNEL_BUG in code, or indication info to reviewers that they should compare by themselves to see if they are KNOWN bugs or they should open NEW bugs. Shall we pick either way,, right? Compare in code if existing issue or let reviewed manually do it but warn them to do so. Otherwise, I am afraid new joiner/people not familiar with the code, would not know what to do with such soft_fail.

Yes, good point. Code is updating and going to be verified.
```
            if (get_var('KNOWN_KERNEL_BUG')) {
                my $failure = "$warn on $machine: " . get_var('KNOWN_KERNEL_BUG');
                $all_failures .= $failure;
                record_soft_failure("$failure and please ignore fake bsc#5555 due to silly PR code rule");
            }
            else {
                record_soft_failure("$warn on $machine: fake bsc#5555. This is an unknown failure and please investiage");
                $all_failures .= "$warn on $machine: unknow failure";
            }
```

> Secondly, bsc#5555 should be a fake bug id, right? Would you please adding `(fake)` after 5555?

Yes, see updates above.


> Last, I remember we talked about how to make such record_soft_fail different from REAL soft_fail? By adding record_info (red borders) around such soft_fail?

There are red failure just following with each soft_fail, fg. https://openqa.suse.de/tests/9948735#step/libvirt_host_bridge_virtual_network/10.  Is it as your expected?

> Maybe a lot comment on this :( , please forgive me . But I do see the importance of raising clear warnings to reviewers to manually check this part and not miss opening new bugs if needed. That's the main purpose of adding the code.

Thanks!

Reply:
https://openqa.suse.de/tests/9948735#step/libvirt_host_bridge_virtual_network/10 is good enough, thanks!

Btw, for the updated code in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/15778/files#r1021061605, will it be able to handle such situation that there are known open bugs already(KNOWN_KERNEL_BUG is not empty), but meanwhile some new coredump/calltrace reported on different module? This won't happen much, but I am fraid code should be able to handle it.

Reply:
No, it needs more effort and more actual cases, and it has lower priority, at present. Anyone can enhance this part when needed.

Reply:
ok, let's accept it as known TODO.